go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -646,6 +654,24 @@ func appendFromRow(pq *sqlparser.ParsedQuery, buf *bytes2.Buffer, fields []*quer
 				buf.WriteString(sqltypes.NullStr)
 			} else {
 				vv := sqltypes.MakeTrusted(typ, row.Values[col.offset:col.offset+col.length])
+
+				if conversion, ok := tp.ConvertCharset[col.field.Name]; ok && col.length >= 0 {
+					// Non-null string value, for which we have a charset conversion instruction
+					fromCollation := tp.CollationEnv.DefaultCollationForCharset(conversion.FromCharset)
```

Comment:
Do we have to rely on the default for the charset? If we take utf8mb4 for example:
```
mysql> show collation where charset = 'utf8mb4';
+----------------------------+---------+-----+---------+----------+---------+---------------+
| Collation                  | Charset | Id  | Default | Compiled | Sortlen | Pad_attribute |
+----------------------------+---------+-----+---------+----------+---------+---------------+
| utf8mb4_0900_ai_ci         | utf8mb4 | 255 | Yes     | Yes      |       0 | NO PAD        |
| utf8mb4_0900_as_ci         | utf8mb4 | 305 |         | Yes      |       0 | NO PAD        |
| utf8mb4_0900_as_cs         | utf8mb4 | 278 |         | Yes      |       0 | NO PAD        |
| utf8mb4_0900_bin           | utf8mb4 | 309 |         | Yes      |       1 | NO PAD        |
| utf8mb4_bg_0900_ai_ci      | utf8mb4 | 318 |         | Yes      |       0 | NO PAD        |
| utf8mb4_bg_0900_as_cs      | utf8mb4 | 319 |         | Yes      |       0 | NO PAD        |
| utf8mb4_bin                | utf8mb4 |  46 |         | Yes      |       1 | PAD SPACE     |
...
| utf8mb4_turkish_ci         | utf8mb4 | 233 |         | Yes      |       8 | PAD SPACE     |
| utf8mb4_unicode_520_ci     | utf8mb4 | 246 |         | Yes      |       8 | PAD SPACE     |
| utf8mb4_unicode_ci         | utf8mb4 | 224 |         | Yes      |       8 | PAD SPACE     |
| utf8mb4_vietnamese_ci      | utf8mb4 | 247 |         | Yes      |       8 | PAD SPACE     |
| utf8mb4_vi_0900_ai_ci      | utf8mb4 | 277 |         | Yes      |       0 | NO PAD        |
| utf8mb4_vi_0900_as_cs      | utf8mb4 | 300 |         | Yes      |       0 | NO PAD        |
| utf8mb4_zh_0900_as_cs      | utf8mb4 | 308 |         | Yes      |       0 | NO PAD        |
+----------------------------+---------+-----+---------+----------+---------+---------------+
89 rows in set (0.00 sec)
```

Reply:
If you're up for squeezing another change in here... I think it should be `CollationConversion` that we use in OnlineDDL. The collation is specific, and it implies the character set.

Reply:
> Do we have to rely on the default collation for the charset (on from and to side)? If we take utf8mb4 for example:

It's a bit moot. We only use `Collation` as an intermediate step to get from the named charset (e.g. `"latin1"`) into a `Charset` object. So we may as well use the default collection to get there.

Reply:
> Perhaps we truly only care about the character set in this scenario though... 🤔

This is worth digging into. If we do end up using collation rather than charset, then there's a few proto changes to make, so this will be outside the scope of this PR.